src/server/engine_shard.cc

```diff
@@ -198,6 +198,33 @@ optional<uint32_t> GetPeriodicCycleMs() {
   return clock_cycle_ms;
 }
 
+/* Eviction begins when the shard's free memory falls below (max_memory_limit *
+ * kEvictionFreeMemoryThreshold) / <number of shards> */
+constexpr double kEvictionFreeMemoryThreshold = 0.1;
+/* or eviction starts if the used RSS memory exceeds max_memory_limit * kEvictionRssThreshold */
+constexpr double kEvictionRssThreshold = 1.0 - kEvictionFreeMemoryThreshold;
```

Comment:
I think this constant is kind of redundant
you can do the same for rss as we do for used memory
rss_limit = rss_oom_deny_ratio * max_memory_limit
eviction_rss_threshold = rss_limit * kEvictionFreeMemoryThreshold
curr_free_rss = rss_limit - rss_mem_current
if (curr_free_rss < eviction_rss_threshold) {
 return eviction_rss_threshold - curr_free_rss
}
return 0